### ISSUE TITLE
es6: let declaration in nested bracless block

### ISSUE THREAD

**Issue by danieljoppi** [spalger/gulp-jshint#122]
```javascript
f(true) 
  for (let i=0; i<10; i++) {
    let a = 'opa'
    console.log(i, a)
  }
```
->

hint.js: line 2, col 8, Let declaration not directly within block.
hint.js: line 3, col 5, Let declaration not directly within block.
hint.js: line 2, col 17, 'i' is not defined.
hint.js: line 2, col 23, 'i' is not defined.
hint.js: line 4, col 17, 'i' is not defined.
hint.js: line 4, col 20, 'a' is not defined.
adding braces fixes the errors:

```javascript
if(true) {
  for (let i=0; i<10; i++) {
    let a = 'opa'
    console.log(i, a)
  }
}
```
edit: de-nesting also fixes the errors:

```javascript
for (let i=0; i<10; i++) {
   let a = 'opa'
   console.log(i, a)
}
```

**Comment by spalger** [comment]
@danieljoppi please report linting errors to https://github.com/jshint/jshint

**Comment by danieljoppi** [comment]
@spalger  was fixed in 2.9.0 [a8fc16b](https://github.com/jshint/jshint/commit/a8fc16b) ... you need update your depencendy
